app/commands/v2/put_link_set.rb

```diff
@@ -9,7 +9,21 @@ def call
           version.increment
           version.save! if link_set.valid?
 
-          link_set.links = merge_links(link_set.links, link_params.fetch(:links))
+          links_hash = link_params.fetch(:links)
+
+          links_hash.each do |link_type, target_content_ids|
+            if target_content_ids.empty?
+              link_set.links.where(link_type: link_type).delete_all
+            else
+              old_target_ids = link_set.links.where(link_type: link_type).pluck(:target_content_id)
+
+              ids_to_be_deleted = old_target_ids - target_content_ids
+              link_set.links.where(link_type: link_type, target_content_id: ids_to_be_deleted).delete_all
+
+              ids_to_be_created = target_content_ids - old_target_ids
+              link_set.links.where(link_type: link_type, target_content_id: ids_to_be_created).create
+            end
+          end
```

Comment:
Rather than moving only the diff'ing logic to the model layer, I'd move the whole link creation/update/delete within it. Maybe inside a method like:

```
def create_or_update_link_items(links, delete_old=false)
      ...
end
```

The method could then be called with `delete_all = true` here:
https://github.com/alphagov/publishing-api/pull/137/files#diff-f618486a57af99708e95cf5c3c2bae66R81

And with `delete_old=false` here in `put_link_set.rb`